### PR TITLE
support metrics in notebook-controller

### LINKED ISSUE #4086: Prometheus/Telemetry metrics for Notebooks Manager UI
/kind feature

**Why you need this feature:**
Identify and create prometheus metrics for the Notebooks Manager UI. This issue is part of getting Notebooks Manager UI ready for Kubeflow 1.0

**Describe the solution you'd like:**
Idenitfy which metrics make sense to track the usage and usability of the Notebooks manager UI. Examples include:

   - #successful notebook creation
   - #notebook pods running across all namespaces, per namespace
   - #failures in pod deployment
   - #reused volume mounts
  and others.

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>
This add some metrics to notebook controller itself, but not add any metrics to manager ui.  So I cannot say it fixes #4086. 
@abhi-g  @lluunn

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4123)

### CODE REVIEW

**Comment by jlewi** [comment]
/ok-to-test
/assign @nrchakradhar  
/assign @kunmingg 

@nrchakradhar can you please review this PR?

**Comment by nrchakradhar** [comment]
@yeya24 Can you please rebase and submit the changes?
From my end the changes are fine. If pre-submit passes this can be merged.

**Comment by yeya24** [comment]
@nrchakradhar  Thanks, I will check the tests.

**Comment by yeya24** [comment]
@nrchakradhar  CI passed.

File: components/notebook-controller/pkg/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	appsv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+// Metrics includes metrics used in notebook controller
+type Metrics struct {
+	cli                  client.Client
+	runningNotebooks     *prometheus.GaugeVec
+	NotebookCreation     *prometheus.CounterVec
```
Comment (lluunn) [inline components/notebook-controller/pkg/metrics/metrics.go]: should we provide a method, and not expose the metrics directly?

Comment (yeya24) [reply components/notebook-controller/pkg/metrics/metrics.go]: Not necessary I believe

Comment (lluunn) [reply components/notebook-controller/pkg/metrics/metrics.go]: I guess we can refactor later if needed

**Comment by lluunn** [comment]
LGTM, thanks for the PR.
You need to sync master though

**Comment by abhi-g** [comment]
👍

**Comment by yeya24** [comment]
> LGTM, thanks for the PR.
> You need to sync master though

Done

File: components/notebook-controller/pkg/metrics/metrics.go (hunk 1 above)
Comment (yeya24) [reply components/notebook-controller/pkg/metrics/metrics.go]: I think maybe we will add more metrics for nb-controller. Maybe how many notebooks are culled or some other metrics. So it is not that easy to wrap these into functions. WDYT?

**Pull request merged**